frontend/src/components/TitleBar/Canvas/CanvasCalendar.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { ReactElement, Component } from 'react';
+import settingStyles from '../Settings/SettingsPage.module.css';
+import styles from './CanvasCalendar.module.css';
+import { setCanvasCalendar, readCanvasCalendar } from '../../../firebase/actions';
+
+type State = {
+  canvasCalendar: string | null;
+  linked: boolean;
+}
+
+export default class CanvasCalendar extends Component<{}, State> {
+  public state: State = {
+    canvasCalendar: '',
+    linked: readCanvasCalendar() != null,
```

Comment:
This is problematic here. You can potentially lose sync with remote canvas calendar field.
You should subscribe to redux store instead.

Subscription code:
https://github.com/cornell-dti/samwise/blob/ee0c0e22ad80895322528003f4f732e9a3270b7b/frontend/src/components/TitleBar/Settings/SettingsPage.tsx#L130-L131

Usage:
(Subscribed props becomes normal props.)
https://github.com/cornell-dti/samwise/blob/ee0c0e22ad80895322528003f4f732e9a3270b7b/frontend/src/components/TitleBar/Settings/SettingsPage.tsx#L62-L69